sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java

```diff
@@ -306,9 +308,27 @@ private static Item create(ItemSpec<?> spec, Path path) {
           spec.getValue(), spec.getShortValue(), spec.getLabel(), spec.getLinkUrl());
     }
 
+    /** <b>For internal use only. No backwards compatibility guarantees.</b> */
+    @Internal
+    public static Item raw(
+        Path path, String key, String value, String shortValue, String label, String linkUrl) {
+      checkNotNull(path, "path cannot be null");
+      checkNotNull(key, "key cannot be null");
+      return new AutoValue_DisplayData_Item(
+          path, Object.class, key, Type.STRING, value, shortValue, label, linkUrl);
+    }
+
     @Override
     public String toString() {
-      return String.format("%s%s:%s=%s", getPath(), getNamespace().getName(), getKey(), getValue());
+      return MoreObjects.toStringHelper(this)
```

Comment:
This is just for pretty-printing for tests. I believe the existing hand-crafted format was to keep it concise since many unit tests involve DisplayData with many items, so a multi-line format can get verbose quick.

`null` is treated as an invalid value for DisplayData; the builder interface has APIs like `.addIfNotNull(myKey, myValue)`. So presumably we shouldn't have to worry about nulls in the proto representation either.